linode/manager
# chore: [M3-6497] - Remove GA and clean up custom events

## Description 📝
**Brief description explaining the purpose of the changes**
Final GA -> AA migration PR with the clean up to remove GA from Cloud Manager. Sorry about commit history on this one. 😅

No target date has been set for this to go into prod yet, but it will at least be after 5/15, when #9004 is released. Until then, this PR can be reviewed but will remain labeled Do Not Merge.

TODO:
- [ ] Merge corresponding internal PR for removing `REACT_APP_GA_ID` and `REACT_APP_GTM_ID`

## Major Changes 🔄
**List highlighting major changes**
- Renames files from `src/utilities/ga` and `src/utilities/ga.test.ts` to `src/utilities/analytics.ts` and `src/utilities/analytics.test.ts`
- Removes `react-ga` as a package
- Removes references to `GA_ID` and `GTM_ID` environment variables
- Removes unused custom events.
- Restores firing of the `sendCreateBucketEvent` when an OBJ storage bucket is created -- we used to track this, but it seems to have gotten dropped when refactoring components. 

## How to test 🧪
1. **How to setup test environment?**
   - Have the `REACT_APP_GA_ID` set in your local `.env`.
   - Have the `REACT_APP_ADOBE_ANALYTICS_URL` set in your local `.env`.
   - `yarn up`
4. **How to verify changes?**
   - Refer to `src/utilities/analytics.ts` for custom events to test.
   - Test actions that fire custom events and verify in the browser console that (1) no Google Analytics data is logged and (2) Adobe Analytics data is logged. See #9004 for details instructions on setting up + expected output for Adobe logging.
6. **How to run Unit or E2E tests?**
 ```yarn cy:run```
